pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercuryOrderDao.java

```diff
@@ -65,9 +67,17 @@ public class MercuryOrderDao implements Dao<MercuryOrderDto> {
             "UPDATE ddp_mercury_sequencing SET order_status = ?, status_date = ?, mercury_pdo_id = ?,  "
                     + " status_detail = ?, status_message = ? where order_id = ?";
 
+    String errorMessage = "Error getting possible mercury orders for participant %s";
 
-    public static void updateOrderStatus(BaseMercuryStatusMessage baseMercuryStatusMessage, String msgData) throws Exception {
+    private static final int MAX_CHAR_ALLOWED = 2000;
+
+
+    public static void updateOrderStatus(BaseMercuryStatusMessage baseMercuryStatusMessage, String msgData) throws DsmInternalError {
         long statusDate = System.currentTimeMillis();
+        //check the length of fields to make sure they fit in the database
+        String messageString = msgData.substring(0, Math.min(msgData.length(), MAX_CHAR_ALLOWED));
```

Comment:
2000 seems generous for the message, esp since we provide a large details section.

Reply:
So here messageString is the json we received from the Mercury. We keep that so if for any reason we miss the fields or something goes wrong (like the current error that was happening) we can always trace back to what was the actual message that we received. If it comes with a large details section we even need to keep more. 
I looked at the last huge message we received and that was too big for our columns that are set to TEXT type, so I thought 2000 would be a good balance. But I am more than willing to change. What number were you thinking?

Reply:
The `msgData` is the whole json of the message that was received, and `baseMercuryStatusMessage` is the part that is related to this order and is parsed into a class.  `details` is just a field in there, which has exactly what you said, audit level details. 

We keep the message that was received in its original form for 2 reasons

1. in case DSM or Mercury break we don't want messages to get lost after acking pubsub topic
2. in case we need to go back and look at the structure of what was received (there were cases when we got concatenated messages from past added to the current message and everything was breaking and DSM was blamed for it, and having a record of what was passed to pubsub and what was received really helped )

In this current error, details was a huge string and message size grew because of that, but there were cases (like the example in the second bullet point ) where message was huge but details had a normal size.